tests/phpunit/api/v3/ChContribution/CreateTest.php

```diff
@@ -26,24 +37,121 @@ public function setUpHeadless() {
    */
   public function setUp() {
     parent::setUp();
+    // If the ch_fund option group is no longer present (likely from the removal of the custom field in the taredown. recreate it
+    $optionGroup = $this->callAPISuccess('OptionGroup', 'get', ['name' => 'ch_fund']);
+    if (empty($optionGroup['count'])) {
+      $optionGroupNew = $this->callAPISuccess('OptionGroup', 'create', [
+        'title' => 'CH Fund',
+        'name' => 'ch_fund',
+        'data_type' => 'String',
+        'description' => '',
+        'is_active' => 1,
+        'is_reserved' => 1,
+      ]);
+      // Ensure that the civicrm_managed table is also updated just in case.
+      CRM_Core_DAO::executeQuery("UPDATE civicrm_managed SET entity_id = %1 WHERE entity_type = 'OptionGroup' AND module = 'biz.jmaconsulting.chfunds'", [1 => [$optionGroupNew['id'], 'Positive']]);
+    }
+    $optionGroup = $this->callAPISuccess('OptionGroup', 'get', ['name' => 'ch_fund']);
+    $this->customGroup = $this->callAPISuccess('CustomGroup', 'create', [
+      'title' => 'Additional info',
+      'extends' => 'Contribution',
+      'collapse_display' => 1,
+      'is_public' => 1,
+      'is_active' => 1,
+    ]);
+    $this->customField = $this->callAPISuccess('CustomField', 'create', [
+      'custom_group_id' => $this->customGroup['id'],
+      'label' => 'CH Fund',
+      'name' => 'Fund',
```

Comment:
@seamuslee001 this might cause an issue if the unit test is executed in a system where we already have a these fields create. Why don't we call a CustomField.Get, OptionGroup.Get API to fetch the id on basis of name 'ch_fund' to be on safe side. And later use the ID in $params array, so if found then it will act like update if not then create. Then in teardown remove

```
    $this->callAPISuccess('CustomField', 'delete', ['id' => $this->customField['id']]);
```
as it will be necessary.

Reply:
@monishdeb probably maybe only call delete if we had created within the extension?